pkg/operation/botanist/health_check_test.go

```diff
@@ -646,14 +653,50 @@ var _ = Describe("health check", func() {
 				},
 			},
 			beConditionWithStatusAndMsg(gardencorev1beta1.ConditionFalse, "MissingNodes", fmt.Sprintf("Not enough worker nodes registered in worker pool '%s' to meet minimum desired machine count. (%d/%d).", workerPoolName2, 0, 1))),
+		Entry("too old Kubernetes patch version",
+			[]*corev1.Node{
+				newNode(nodeName, true, labels.Set{"worker.gardener.cloud/pool": workerPoolName1}, "v1.19.2"),
+			},
+			[]gardencorev1beta1.Worker{
+				{
+					Name:    workerPoolName1,
+					Maximum: 10,
+					Minimum: 1,
+				},
+			},
+			beConditionWithStatusAndMsg(gardencorev1beta1.ConditionFalse, "KubeletVersionMismatch", fmt.Sprintf("The kubelet version for node %q (v1.19.2) does not match the desired Kubernetes version (v%s)", nodeName, kubernetesVersion.Original()))),
+		Entry("same Kubernetes patch version",
+			[]*corev1.Node{
+				newNode(nodeName, true, labels.Set{"worker.gardener.cloud/pool": workerPoolName1}, "v1.19.3"),
+			},
+			[]gardencorev1beta1.Worker{
+				{
+					Name:    workerPoolName1,
+					Maximum: 10,
+					Minimum: 1,
+				},
+			},
+			BeNil()),
+		Entry("different Kubernetes minor version (all healthy)",
+			[]*corev1.Node{
+				newNode(nodeName, true, labels.Set{"worker.gardener.cloud/pool": workerPoolName1}, "v1.18.2"),
+			},
+			[]gardencorev1beta1.Worker{
+				{
+					Name:    workerPoolName1,
+					Maximum: 10,
+					Minimum: 1,
+				},
+			},
+			BeNil()),
```

Comment:
That should fail right?

Why would the condition be READY if the worker node still has a different minor version as the Shoot?

Reply:
A rolling update might take time, so there is no reason to fail the condition. It's completely fine and normal to have two sets of nodes with different minor versions during such an upgrade.

Reply:
If there is minor version mismatch, it should be detected by the worker controller.